### PR TITLE
Settings upgrade

### PR DESCRIPTION
**Pull request by PowerOfNames**
This PR makes some changes to the Settings.kt class, adding save and load methods to work with .properties files.
In addition to that, adds the SettingsEditor and SettingsEditorExample files.

The SettingsEditor is a Swing based UI, that presents a loaded setting in a 2 column table, showing the settings name plus its value.
The UI can be used to add new settings, save and load the setting into a .properties file, and refresh the shown settings (to present value changed that could happen during runtime)

The SettingEditorExample is an adaptation of the TexturedCubeExample and shows how to set and reset settings values, aand how to add an updateRoutine to a specific setting (which is called when the setting value changes).

Disclaimer: Same as for the TransferFunctionEditor, the Swing window might be tricky to test.

### CODE REVIEW

**Review by moreApi — APPROVED** [review]
Looks good to me. Now its time for @skalarproduktraum

File: src/main/kotlin/graphics/scenery/SettingsEditor.kt [hunk 1]
```diff
@@ -0,0 +1,226 @@
...
+import javax.swing.table.DefaultTableModel
+
+/**
+ * @author Konrad Michel <konrad.michel@mailbox.tu-dresden.de>
+ *
+ * A Swing UI, that allows the user to load setting in order to manipulate and add settings.
+ * Allows saving and loading of settings into a .properties file
+ * Allows refreshing of the settings table in order to show changes to settings values that happened during runtime.
+ */
+class SettingsEditor(var settings : Settings, private val mainFrame : JFrame = JFrame("SettingsEditor"), width : Int = 480, height : Int = 500) {
```
Comment (moreApi) [inline src/main/kotlin/graphics/scenery/SettingsEditor.kt:22]: please put "@JvmOverloads" in line 21 so it is more convinient to use in Java

**Review by moreApi — CHANGES_REQUESTED** [review]

File: src/main/kotlin/graphics/scenery/Settings.kt [hunk 2]
```diff
@@ -9,30 +13,93 @@ import java.util.concurrent.ConcurrentHashMap
  *
  * @author Ulrik Günther <hello@ulrik.is>
  */
-class Settings(override var hub: Hub? = null) : Hubable {
+class Settings(override var hub: Hub? = null, prefix : String = "scenery.", propertiesFilePath : String? = null) : Hubable {
```
Comment (skalarproduktraum) [inline src/main/kotlin/graphics/scenery/Settings.kt:16]: Might be worth making `prefix` a val so it could be queried later

File: src/main/kotlin/graphics/scenery/Settings.kt [hunk 3]
```diff
@@ -9,30 +13,93 @@ import java.util.concurrent.ConcurrentHashMap
...
     private var settingsStore = ConcurrentHashMap<String, Any>()
     private val logger by LazyLogger()
 
+    var settingsUpdateRoutines : HashMap<String, ArrayList<() -> Unit>> = HashMap()
```
Comment (skalarproduktraum) [inline src/main/kotlin/graphics/scenery/Settings.kt:20]: This should probably be private so it can't be fiddled with externally.

File: src/main/kotlin/graphics/scenery/Settings.kt [hunk 4]
```diff
@@ -9,30 +13,93 @@ import java.util.concurrent.ConcurrentHashMap
...
+    lateinit var propertiesFile : File
```
Comment (skalarproduktraum) [inline src/main/kotlin/graphics/scenery/Settings.kt:21]: I think it might make sense not to give a path as `String` to the class, but instead an InputStream. This way, the properties could be read from whereever, JAR file, URL, etc. That would also remove the necessity for the lateinit var here.

File: src/main/kotlin/graphics/scenery/SettingsEditor.kt (hunk 1 above)
Comment (skalarproduktraum) [reply src/main/kotlin/graphics/scenery/SettingsEditor.kt:22]: If confused on how to add this annotation here, see `SceneryBase` for an example 👍

**Review by skalarproduktraum — CHANGES_REQUESTED** [review]
Looks great, I just had a few little things to complain about. Address them and we're good to go! Thanks @PowerOfNames :+1:

File: src/main/kotlin/graphics/scenery/Settings.kt [hunk 5]
```diff
@@ -9,30 +13,93 @@ import java.util.concurrent.ConcurrentHashMap
...
+        else
+            out = FileOutputStream(propertiesFile)
+
+        props.store(out, null)
+    }
+
+    /**
+     * Parses the type from the incoming string, returns the casted value
+     */
+    fun parseType(value:String): Any = when {
```
[the remark this answers is not in the document]
Comment (PowerOfNames) [reply src/main/kotlin/graphics/scenery/Settings.kt:95]: I cant be private because I use it in other places too. But I ned to tag it that its used somewhere else (In Microscenery, I need to to parcing there) An alternative would be to move it into a Utilities class, but I think thats not your way to do things with scenery :D

File: src/main/kotlin/graphics/scenery/Settings.kt (hunk 4 above)
Comment (PowerOfNames) [reply src/main/kotlin/graphics/scenery/Settings.kt:21]: I c. I wasnt aware that a InpoutStream is a hashable object

File: src/main/kotlin/graphics/scenery/SettingsEditor.kt (hunk 1 above)
Comment (PowerOfNames) [reply src/main/kotlin/graphics/scenery/SettingsEditor.kt:22]: 'This annotation is not applicable to target 'class'' -> Also did not do that in the TransferFunctionEditor?!

File: new.properties [hunk 6]
```diff
@@ -0,0 +1,4 @@
+#Wed Feb 01 16:44:23 CET 2023
```
Comment (skalarproduktraum) [inline new.properties:1]: Is this file needed?

File: newnew.properties [hunk 7]
```diff
@@ -0,0 +1,5 @@
+#Wed Feb 01 16:33:04 CET 2023
```
Comment (skalarproduktraum) [inline newnew.properties:1]: Is this file needed?

**Review by skalarproduktraum — CHANGES_REQUESTED** [review]
Looks good, please have a look at the properties files though. They should either be not there, or in a different place (resources folder, if needed for a test, etc).

File: newnew.properties (hunk 7 above)
Comment (PowerOfNames) [reply newnew.properties:1]: whoops, they are not supposed to be there

**Review by skalarproduktraum — APPROVED** [review]
